atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -516,6 +534,52 @@ private void unmarkStreamsAsUsed() {
                     line("index.delete(toDelete);");
                 } line("}");
             }
+
+            private void storeBlocksAndGetFinalMetadata() {
+                line("@Override");
+                line("protected StreamMetadata storeBlocksAndGetFinalMetadata(Transaction t, long id, InputStream stream) {"); {
+                    line("//Hash the data before compressing it");
+                    line("MessageDigest digest = Sha256Hash.getMessageDigest();");
+                    line("InputStream hashingStream = new DigestInputStream(stream, digest);");
```

Comment:
ideally yes, though in reality it will be closed when the `compressingStream` is closed, and a `DigestInputStream` doesn't currently do anything on close (though it could in the future).